### PR TITLE
:sparkles: Create Recipes and saving to DB; Tests

### LINKED ISSUE #20: Recipe is saved to DB

### PR DESCRIPTION
**Pull request by Jothaka**
Closes #20 
Recipes can be created and added to the DB with a POST on ("api/recipe").
Recipes have a unique ID.
Unit tests for RecipeService.saveRecipe and IdUtility.generateID
Integration test for POST mapping

### CODE REVIEW

File: backend/src/main/java/de/jankahle/capstone/service/RecipeService.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    @Autowired
+    public RecipeService(RecipeMongoDB recipeMongoDB) {
+        this.recipeMongoDB = recipeMongoDB;
+    }
+
+    public Recipe saveRecipeToDB(RecipeDto recipeDto){
+
+        Recipe recipe = recipeDto.toRecipe();
+        recipe.setId(IdUtility.generateId());
```
Comment (lisamittmann) [inline backend/src/main/java/de/jankahle/capstone/service/RecipeService.java:23]: you do not check if recipes already exist - if this is intended all is fine, if not, something to keep in mind

File: backend/src/test/java/de/jankahle/capstone/controller/RecipeControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    public  void setup() {
+        recipeMongoDB.deleteAll();
+    }
+
+    @Test
+    @DisplayName("Saving a new Recipe adds the Recipe to the DB")
+    void saveRecipe() {
+        //GIVEN
+        List<Ingredient> ingredients = List.of(
+                Ingredient.builder().name("Kartoffel").amount(800).measurementUnit("gram").build(),
```
Comment (lisamittmann) [inline backend/src/test/java/de/jankahle/capstone/controller/RecipeControllerTest.java:50]: Gramm ;)

File: backend/src/test/java/de/jankahle/capstone/service/RecipeServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class RecipeServiceTest {
+
+    private final RecipeMongoDB recipeMongoDB = mock(RecipeMongoDB.class);
```
Comment (lisamittmann) [inline backend/src/test/java/de/jankahle/capstone/service/RecipeServiceTest.java:19]: mock is not used in test, you may want to remove it or include a mock response within your test

**Review by lisamittmann — APPROVED** [review]
lgtm, just some minor pointers

File: backend/src/test/java/de/jankahle/capstone/service/RecipeServiceTest.java (hunk 3 above)
Comment (Jothaka) [reply backend/src/test/java/de/jankahle/capstone/service/RecipeServiceTest.java:19]: It actually IS used in the test, but the implementation does not matter in the context of the test as the return value of the recipeMongoDB.save method is not used in any way. Therefore we do not need to mock with when.
I'll add a verify to check if the save method is invoked though.

File: backend/src/main/java/de/jankahle/capstone/service/RecipeService.java (hunk 1 above)
Comment (Jothaka) [reply backend/src/main/java/de/jankahle/capstone/service/RecipeService.java:23]: is intended :)

**Pull request merged**